File: rust/remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        //nums -> iter -> dereference elements -> filter elements -> collect
        *nums = nums.into_iter().map(|n| *n).filter(|n| *n != val).collect();

        nums.len() as i32
    }
}

File: c++/remove-element.cpp
#include <vector>
using std::vector;
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int j = 0;
        int size = nums.size();
        for(int i = 0; i < size; i++){
            if(nums[i] != val){
                nums[j] = nums[i];
                j++;
            }
        }
        return j;
    }
};